snapcraft/plugins/python.py

```diff
@@ -0,0 +1,237 @@
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - requirements:
+      (string)
+      Path to a requirements.txt file
+    - constraints:
+      (string)
+      Path to a constraints file
+    - process-dependency-links:
+      (bool; default: false)
+      Enable the processing of dependency links.
+    - python-packages:
+      (list)
+      A list of dependencies to get from PyPi
+    - python-version:
+      (string; default: python3)
+      The python version to use. Valid options are: python2 and python3
+"""
+
+import os
+import re
+import subprocess
+from glob import glob
+
+import snapcraft
+from snapcraft import common
+
+
+class PythonPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        schema['properties']['requirements'] = {
+            'type': 'string',
+        }
+        schema['properties']['constraints'] = {
+            'type': 'string',
+        }
+        schema['properties']['python-packages'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'default': [],
+        }
+        schema['properties']['process-dependency-links'] = {
+            'type': 'boolean',
+            'default': False,
+        }
+        schema['properties']['python-version'] = {
+            'type': 'string',
+            'default': 'python3',
+            'enum': ['python2', 'python3']
+        }
+        schema.pop('required')
+
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the pull step dirty.
+        schema['pull-properties'].extend([
+            'requirements',
+            'constraints',
+            'python-packages',
+            'python-version',
+        ])
+
+        return schema
+
+    @property
+    def plugin_build_packages(self):
+        if self.options.python_version == 'python3':
+            return [
+                'python3-dev',
+                'python3-pip',
+                'python3-pkg-resources',
+                'python3-setuptools',
+            ]
+        elif self.options.python_version == 'python2':
+            return [
+                'python-dev',
+                'python-pip',
+                'python-pkg-resources',
+                'python-setuptools',
+            ]
+        else:
+            raise AssertionError('python-version has a value of {!r}'.format(
+                self.options.python_version))
+
+    @property
+    def plugin_stage_packages(self):
+        if self.options.python_version == 'python3':
+            return ['python3']
+        elif self.options.python_version == 'python2':
+            return ['python']
+        else:
+            raise AssertionError('python-version has a value of {!r}'.format(
```

Comment:
Hmmm. I suppose schema could be added to the children too if that a problem.

I just don't like the idea of checking python versions in multiple places. Perhaps abstract the else to another function so we don't have to change exceptions in multiple places should this need to be handled differently in the future? (I'm seeing potential uses for python3.4 vs python3.5 maybe). I just don't like violating DRY in this way.